src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -986,6 +1012,76 @@ protected override int ExecuteCmd(DocumentsOperationContext context)
+                            var clonedDocData = existingDoc.Data.Clone(context);
+                            
+                            if (existingDoc.Flags.Contain(DocumentFlags.HasRevisions) == false)
+                            {
+                                // When forcing a revision for a document that doesn't have revisions yet, 
+                                // we must add HasRevisions flag to the document and save (put)  
+                                existingDoc.Flags = existingDoc.Flags |= DocumentFlags.HasRevisions;
+                                
+                                putResult = Database.DocumentsStorage.Put(context, existingDoc.Id,
+                                                                         existingDoc.ChangeVector, 
+                                                                          clonedDocData, 
+                                                                          flags: existingDoc.Flags,
+                                                                          nonPersistentFlags: NonPersistentDocumentFlags.SkipRevisionCreation);
+
+                                existingDoc.ChangeVector = putResult.ChangeVector;
+                                existingDoc.LastModified = putResult.LastModified;
+                            }
+                            
+                            // Create the revision. Revision will be created only if there isn't a revision with identical content for this document  
+                            revisionCreated = Database.DocumentsStorage.RevisionsStorage.Put(context, existingDoc.Id, 
+                                                                                         clonedDocData, 
+                                                                                         existingDoc.Flags, 
+                                                                                         nonPersistentFlags: NonPersistentDocumentFlags.None,
+                                                                                         existingDoc.ChangeVector,
+                                                                                         existingDoc.LastModified.Ticks);
+                            if (revisionCreated)
+                            {
+                                // Reply with new revision data
+                                forceRevisionReply = new DynamicJsonValue
+                                {
+                                    ["RevisionCreated"] = true,
+                                    ["Type"] = nameof(CommandType.ForceRevisionCreation), 
+                                    [Constants.Documents.Metadata.Id] = existingDoc.Id,
+                                    [Constants.Documents.Metadata.Collection] =  CollectionName.GetCollectionName(existingDoc.Data),
+                                    [Constants.Documents.Metadata.ChangeVector] = existingDoc.ChangeVector,
```

Comment:
I'm wondering if we need this actually. What we are going to do with this on client-side?

Reply:
I don't think we _can_ do anything with it. There is no record of revisions on the session, and no need to track those.
I guess we might want to save the change vector of the revision? But we aren't returning it

Reply:
We're tracking the document in the session.
When the document doesn't have the "HasRevisions" flag we need to update it and return the new document change vector.